src/map/core.rs

```diff
@@ -8,28 +8,42 @@
 //! However, we should probably not let this show in the public API or docs.
 
 mod entry;
-mod raw;
 
 pub mod raw_entry_v1;
 
-use hashbrown::raw::RawTable;
+use hashbrown::hash_table;
 
 use crate::vec::{self, Vec};
 use crate::TryReserveError;
 use core::mem;
 use core::ops::RangeBounds;
 
 use crate::util::simplify_range;
-use crate::{Bucket, Entries, Equivalent, HashValue};
+use crate::{Bucket, Equivalent, HashValue};
+
+type Indices = hash_table::HashTable<usize>;
+type Entries<K, V> = Vec<Bucket<K, V>>;
 
 pub use entry::{Entry, IndexedEntry, OccupiedEntry, VacantEntry};
 
 /// Core of the map that does not depend on S
+#[derive(Debug)]
 pub(crate) struct IndexMapCore<K, V> {
     /// indices mapping from the entry hash to its index.
-    indices: RawTable<usize>,
-    /// entries is a dense vec of entries in their order.
-    entries: Vec<Bucket<K, V>>,
+    indices: Indices,
+    /// entries is a dense vec maintaining entry order.
+    entries: Entries<K, V>,
+}
+
+/// Mutable references to the parts of an `IndexMapCore`.
+///
+/// When using `HashTable::find_entry`, that takes hold of `&mut indices`, so we have to borrow our
+/// `&mut entries` separately, and there's no way to go back to a `&mut IndexMapCore`. So this type
```

Comment:
@clarfonthey Just to highlight, this comment is the part that I find most unfortunate about the change: it makes our `OccupiedEntry` larger to carry separate `&mut` references, and also makes the core methods more awkward to call from the entry. This new `RefMut` recovers some of the ergonomics, but I don't know that it's possible to avoid the size increase at all.

Before, `raw::Bucket` allowed me to store that with the full `&mut IndexMapCore` in the entry.